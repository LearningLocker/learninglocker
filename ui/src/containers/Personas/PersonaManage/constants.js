export const identifierTypeDisplays = {
  mbox: 'Email',
  mbox_sha1sum: 'Encrypted Email',
  openid: 'Open ID',
  account: 'Website Account',
};

export default { identifierTypeDisplays };
