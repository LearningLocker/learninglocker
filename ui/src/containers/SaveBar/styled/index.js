import styled from 'styled-components';
import ProgressBar from 'react-toolbox/lib/progress_bar';
import { compose, withProps } from 'recompose';

// Theme variables to override react toolbox component's styling
const valueClassName = 'value';
const linearClassName = 'linear';

export const Container = styled.div`
  margin-right: -11px;
  top: 56px;
  height: 0.4rem;
  position: relative;
  z-index: 5;

  .${linearClassName} {
    background: none;
  }

  .${valueClassName} {
    height: 0.4rem;
  }
`;

const getSavingStageColor = (savingStageClassName) => {
  switch (savingStageClassName) {
    case 'failed':
      return '#ff0000';
    case 'completed':
      return '#3ecb7a';
    default:
      return 'rgb(245, 171, 53)';
  }
};

export const Label = styled.div`
  position: absolute;
  top: 0.4rem;
  height: 2rem;
  left: 0;
  right: 0;
  text-align: center;
  font-weight: bold;
  ${props => `color: ${getSavingStageColor(props.savingStageClassName)};`}
`;

export const StyledProgressBar = compose(
  withProps({
    theme: {
      linear: linearClassName,
      value: valueClassName,
    }
  })
)(
  styled(ProgressBar)`
    top: -11px;

    .${valueClassName} {
      ${props => `background-color: ${getSavingStageColor(props.savingStageClassName)};`}
    }

    .${linearClassName} {
      display: block;
    }
  `
);
