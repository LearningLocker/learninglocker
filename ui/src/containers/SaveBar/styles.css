.value {
  background-color: rgb(245, 171, 53);
}

.linear {
  display: block;
}

.container {
  margin-right: -11px;
  top: 56px;
  height: 0.4rem;
  position: relative;
  z-index: 5;
}

.container .linear {
  background: none;
}

.container .value {
  height: 0.4rem;
}

.label {
  color: rgb(245, 171, 53);
  position: absolute;
  top: 0.4rem;
  height: 2rem;
  left: 0px;
  right: 0px;
  text-align: center;
  font-weight: bold;
}

.label.failed {
  color: red;
}

.label.completed {
  color: #3ECB7A;
}

.failed .value {
  background-color: red;
}
.completed .value{
  background-color: #3ECB7A;
}
