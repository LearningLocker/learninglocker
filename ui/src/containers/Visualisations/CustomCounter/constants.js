import { COUNTER_IMAGE } from 'ui/components/VisualiseIcon/assets';

export const title = 'Counter';
export const image = COUNTER_IMAGE;
