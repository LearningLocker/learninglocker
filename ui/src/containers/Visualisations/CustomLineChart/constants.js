import { FREQUENCY_IMAGE } from 'ui/components/VisualiseIcon/assets';

export const title = 'Line';
export const image = FREQUENCY_IMAGE;
