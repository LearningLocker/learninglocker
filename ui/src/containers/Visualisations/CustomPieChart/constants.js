import { PIE_IMAGE } from 'ui/components/VisualiseIcon/assets';

export const title = 'Pie';
export const image = PIE_IMAGE;
