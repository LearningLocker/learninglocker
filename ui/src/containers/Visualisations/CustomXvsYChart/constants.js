import { XVSY_IMAGE } from 'ui/components/VisualiseIcon/assets';

export const title = 'Correlation';
export const image = XVSY_IMAGE;
