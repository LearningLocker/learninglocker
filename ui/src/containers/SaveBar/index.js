import React from 'react';
import { fromJS, List } from 'immutable';
import { createSelector } from 'reselect';
import { connect } from 'react-redux';
import { COMPLETED, FAILED, IN_PROGRESS } from 'ui/utils/constants';
import { compose } from 'recompose';
import SaveBarErrors from 'ui/containers/SaveBarErrors';
import {
  Container,
  Label,
  StyledProgressBar,
} from 'ui/containers/SaveBar/styled';

export const savingSelector = () => createSelector(
  state => // models
    state.models
      .map(
        model =>
          model.filter(item =>
            item && item.getIn &&
            (
              !!item.getIn(['remoteCache', 'requestState']) ||
              !!item.getIn(['deleteState'])
            )
          )
      )
      .toList()
      .flatMap(
        model => model
          .toList()
          .flatMap(
            item =>
              // item && item.getIn && item.getIn(['remoteCache', 'requestState'])
              new List([
                item && item.getIn && item.getIn(['remoteCache', 'requestState']),
                item && item.getIn && item.getIn(['deleteState'])
              ])
          )
      )
  ,
  state => // uploadPersonas
    state
      .uploadPersonas
      .filter(model => model && model.getIn && !!model.getIn(['requestState']))
      .toList()
      .map(model => model.getIn(['requestState'])),
  state => // uploadPeople
    state
      .mergePersona
      .filter(model => model && model.getIn && !!model.getIn(['requestState']))
      .toList()
      .map(model => model.getIn(['requestState'])),
  state => // update UserOrganisationSettings
    fromJS(state.userOrganisationSettings || {})
      .flatMap((v1, k1) => v1.mapKeys(k2 => `${k1}-${k2}`))
      .filter(model => model && model.getIn && !!model.getIn(['requestState']))
      .toList()
      .map(model => model.getIn(['requestState'])),
  (
    saving,
    uploadPersonasSaving,
    mergePersonaSaving,
    userOrganisationSettingsSaving
  ) => {
    saving = saving
      .concat(uploadPersonasSaving)
      .concat(mergePersonaSaving)
      .concat(userOrganisationSettingsSaving);

    if (saving.includes(IN_PROGRESS)) {
      return IN_PROGRESS;
    }

    if (saving.includes(FAILED)) {
      return FAILED;
    }

    if (saving.includes(COMPLETED)) {
      return COMPLETED;
    }

    return false;
  }
);

const getLabel = (value) => {
  switch (value) {
    case IN_PROGRESS:
      return 'SAVING';
    case COMPLETED:
      return 'SAVED';
    case FAILED:
      return 'SAVE FAILED';
    default:
      return '';
  }
};

const SaveBar = ({
  saving
}) => {
  if (!saving) {
    return null;
  }

  const savingStageClassName = saving.toLowerCase();

  return (
    <Container className={'save-bar'}>
      <StyledProgressBar
        savingStageClassName={savingStageClassName}
        mode={saving === IN_PROGRESS ? 'indeterminate' : 'determinate'}
        value={100} />
      <Label savingStageClassName={savingStageClassName}>
        {getLabel(saving)}
      </Label>
      <SaveBarErrors />
    </Container>);
};

const SaveBarComposed = compose(
  connect(
    state => ({
      saving: savingSelector()(state)
    })
  )
)(SaveBar);

export default SaveBarComposed;
