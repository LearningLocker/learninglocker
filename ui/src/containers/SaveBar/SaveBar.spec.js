import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS, List } from 'immutable';
import { COMPLETED, FAILED, IN_PROGRESS } from 'ui/utils/constants';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import SaveBar, { savingSelector } from './index.js';

const createMockStore = ({
  requestState = IN_PROGRESS
}) => {
  const mockState = {
    models: fromJS({
      modelType: {
        theModelId: {
          remoteCache: {
            requestState
          }
        }
      }
    }),
    uploadPersonas: fromJS({}),
    mergePersona: fromJS({}),
    alerts: new List(),
    userOrganisationSettings: {},
  };

  return createStore((data, action) => {
    switch (action.type) {
      case '@@redux/INIT':
        return data;
      case 'addModel':
        data.models = data.models.setIn(['modelType', action.modelId], fromJS(action.model));
        return data;
      default:
        return;
    }
  }, mockState);
};

describe('SaveBar', () => {
  it('should render save bar IN_PROGRESS', () => {
    const mockStore = createMockStore({ requestState: IN_PROGRESS });

    const sideNav = renderer
      .create(
        <Provider store={mockStore}>
          <SaveBar />
        </Provider>
      )
      .toJSON();

    expect(sideNav).toMatchSnapshot();
  });

  it('should render save bar COMPLETED', () => {
    const mockStore = createMockStore({ requestState: COMPLETED });

    const sideNav = renderer
      .create(
        <Provider store={mockStore}>
          <SaveBar />
        </Provider>
      )
      .toJSON();

    expect(sideNav).toMatchSnapshot();
  });

  it('should render save bar FAILED', () => {
    const mockStore = createMockStore({ requestState: FAILED });

    const sideNav = renderer
      .create(
        <Provider store={mockStore}>
          <SaveBar />
        </Provider>
      )
      .toJSON();

    expect(sideNav).toMatchSnapshot();
  });

  it('reducer should be in progress', () => {
    const mockStore = createMockStore({});

    mockStore.dispatch({
      type: 'addModel',
      modelId: '123',
      model: {
        requestState: FAILED
      }
    });

    mockStore.dispatch({
      type: 'addModel',
      modelId: '456',
      model: {
        requestState: COMPLETED
      }
    });

    const result = savingSelector()(mockStore.getState());

    expect(result).toEqual(IN_PROGRESS);
  });

  it('reducer should be false', () => {
    const mockStore = createMockStore({ requestState: null });

    const result = savingSelector()(mockStore.getState());

    expect(result).toEqual(false);
  });
});
