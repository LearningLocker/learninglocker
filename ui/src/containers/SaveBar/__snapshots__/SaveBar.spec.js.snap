// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveBar should render save bar COMPLETED 1`] = `
<div
  className="sc-bwzfXH gWHEGL save-bar"
>
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={100}
    className="linear sc-bxivhb guboWi"
    data-react-toolbox="progress-bar"
    disabled={undefined}
  >
    <div>
      <span
        className={undefined}
        data-ref="buffer"
        style={
          Object {
            "MsTransform": "scaleX(0)",
            "WebkitTransform": "scaleX(0)",
            "transform": "scaleX(0)",
          }
        }
      />
      <span
        className="value"
        data-ref="value"
        style={
          Object {
            "MsTransform": "scaleX(1)",
            "WebkitTransform": "scaleX(1)",
            "transform": "scaleX(1)",
          }
        }
      />
    </div>
  </div>
  <div
    className="sc-htpNat fHEvbJ"
  >
    SAVED
  </div>
  <div
    className="sc-bdVaJa bgbluT"
  />
</div>
`;

exports[`SaveBar should render save bar FAILED 1`] = `
<div
  className="sc-bwzfXH gWHEGL save-bar"
>
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={100}
    className="linear sc-bxivhb kOaAkp"
    data-react-toolbox="progress-bar"
    disabled={undefined}
  >
    <div>
      <span
        className={undefined}
        data-ref="buffer"
        style={
          Object {
            "MsTransform": "scaleX(0)",
            "WebkitTransform": "scaleX(0)",
            "transform": "scaleX(0)",
          }
        }
      />
      <span
        className="value"
        data-ref="value"
        style={
          Object {
            "MsTransform": "scaleX(1)",
            "WebkitTransform": "scaleX(1)",
            "transform": "scaleX(1)",
          }
        }
      />
    </div>
  </div>
  <div
    className="sc-htpNat jvZXLN"
  >
    SAVE FAILED
  </div>
  <div
    className="sc-bdVaJa bgbluT"
  />
</div>
`;

exports[`SaveBar should render save bar IN_PROGRESS 1`] = `
<div
  className="sc-bwzfXH gWHEGL save-bar"
>
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={100}
    className="linear sc-bxivhb fGBlup"
    data-react-toolbox="progress-bar"
    disabled={undefined}
  >
    <div>
      <span
        className={undefined}
        data-ref="buffer"
        style={undefined}
      />
      <span
        className="value"
        data-ref="value"
        style={undefined}
      />
    </div>
  </div>
  <div
    className="sc-htpNat hoqFTd"
  >
    SAVING
  </div>
  <div
    className="sc-bdVaJa bgbluT"
  />
</div>
`;
